src/main/host/memory_manager/mod.rs

```diff
@@ -163,7 +163,7 @@ where
 
     fn deref(&self) -> &Self::Target {
         match &self.0 {
-            CopiedOrMapped::Copied(v) => &v,
+            CopiedOrMapped::Copied(v) => v,
```

Comment:
I'm confused how this works, since `v` is a `Vec`, not a `&Vec`. If it compiles I guess it's right, though...?

Reply:
I think that's because the reference in `&self.0` causes the pattern to be in ref binding mode, so `v` is bound to a reference to the inner type `&Vec` instead of `Vec`.